aeppl/transforms.py

```diff
@@ -89,52 +90,113 @@ def transformed_logprob(op, value, *inputs, name=None, **kwargs):
     return logprob + jacobian
 
 
+class DefaultTransformSentinel:
+    pass
+
+
+DEFAULT_TRANSFORM = DefaultTransformSentinel()
+
+
 @local_optimizer(tracks=None)
-def default_transform(fgraph: FunctionGraph, node: Node) -> Optional[List[Node]]:
+def transform_values(fgraph: FunctionGraph, node: Node) -> Optional[List[Node]]:
     """
-    Apply default transform to value variables. It is assumed that the input
-    value variables correspond to forward transformations, usually chosen in
+    Apply transform to value variables. It is assumed that the input value
+    variables correspond to forward transformations, usually chosen in
     such a way that the values are unconstrained on the real line.
 
     e.g., if Y ~ HalfNormal, we assume the respective value variable is specified
     on the log scale and back-transform it to obtain Y on the natural scale.
     """
 
     rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    values_to_transforms = getattr(fgraph, "values_to_transforms", None)
 
-    if rv_map_feature is None:
-        return None  # pragma: no cover
-
-    if not isinstance(node.op, RandomVariable):
-        return None  # pragma: no cover
-
-    trans_node = _default_transformed_rv(node.op, node)
-
-    if trans_node is None:
+    if rv_map_feature is None or values_to_transforms is None:
         return None  # pragma: no cover
 
-    # Get the old value variable and remove it from our value variables map
-    rv_var = node.outputs[1]
-    old_value_var = rv_map_feature.rv_values.pop(rv_var)
```

Comment:
There was no need to pop the `old_value_var` since we already have it from above and we are replacing it with the `new_value_var` at the end anyway.

Reply:
If we don't pop them, then we'll end up with a map that has entries for variables that are no longer in the graph, and&mdash;unless I'm missing something&mdash;we don't want/need those.

Reply:
The original rewrite was reintroducing it in the dictionary, but associated with the new value. See the other comment.